### PR TITLE
APPS-3316: Course finished screen

### PR DESCRIPTION
**Pull request by rostikjoystick**
**YouTrack task**: [APPS-3273](https://vyahhi.myjetbrains.com/youtrack/issue/APPS-3273),[#APPS-3316](https://vyahhi.myjetbrains.com/youtrack/issue/APPS-3316)

**Description List**:
* Add course finished dialog

### CODE REVIEW

File: app/src/main/java/org/stepik/android/view/course_complete/ui/dialog/CourseCompleteBottomSheetDialogFragment.kt [hunk 1]
```diff
@@ -0,0 +1,620 @@
...
+            courseCompleteTitle,
+            courseCompleteFeedback,
+            courseCompleteSubtitle,
+            viewCertificateAction,
+            shareResultAction,
+            courseCompleteDivider,
+            primaryAction,
+            secondaryAction
+        )
+        viewStateDelegate.addState<CourseCompleteFeature.State.NetworkError>()
```
Comment (eadm) [inline app/src/main/java/org/stepik/android/view/course_complete/ui/dialog/CourseCompleteBottomSheetDialogFragment.kt:115]: Is it ok to ignore error state?

**Review by eadm — COMMENTED** [review]
- ripple should be full width
- points to certificate with distinction not shown
- landscape mode should be handled with scroll
Screenshot_1623184989
Screenshot_1623184943

File: app/src/main/java/org/stepik/android/data/certificate/repository/CertificateRepositoryImpl.kt [hunk 2]
```diff
@@ -16,6 +17,25 @@ constructor(
     private val certificateCacheDataSource: CertificateCacheDataSource,
     private val certificateRemoteDataSource: CertificateRemoteDataSource
 ) : CertificateRepository {
+    override fun getCertificate(userId: Long, courseId: Long, sourceType: DataSourceType): Maybe<Certificate> {
```
[the remark this answers is not in the document]
Comment (rostikjoystick) [reply app/src/main/java/org/stepik/android/data/certificate/repository/CertificateRepositoryImpl.kt:20]: `getCertificate` has two id parameters :disappointed:

File: app/src/main/java/org/stepik/android/view/course_complete/ui/dialog/CourseCompleteBottomSheetDialogFragment.kt (hunk 1 above)
Comment (rostikjoystick) [reply app/src/main/java/org/stepik/android/view/course_complete/ui/dialog/CourseCompleteBottomSheetDialogFragment.kt:115]: Will dismiss dialog and show snackbar if this state is set

**Review by eadm — APPROVED** [review]

**Pull request merged**